packages/connectkit/src/components/DaimoPayButton/index.tsx

```diff
@@ -1,128 +1,217 @@
-    chain?: Chain & {
-      unsupported?: boolean;
-    };
-    unsupported: boolean;
-    isConnected: boolean;
-    isConnecting: boolean;
-    address?: Hash;
-    truncatedAddress?: string;
-    ensName?: string;
-    payId?: string;
+  children: (renderProps: {
+    show: () => void;
+    hide: () => void;
   }) => React.ReactNode;
 };
 
-/** Like DaimoPayButton, but with custom styling. */
-const DaimoPayButtonRenderer: React.FC<DaimoPayButtonRendererProps> = ({
-  payId,
-  closeOnSuccess,
-  children,
-}) => {
-  const isMounted = useIsMounted();
+/**
+ * A button that shows the Daimo Pay checkout. Replaces the traditional
+ * Connect Wallet » approve » execute sequence with a single action.
+ */
+export function DaimoPayButton(props: DaimoPayButtonProps) {
+  const { theme, mode, customTheme, onClick } = props;
   const context = usePayContext();
 
-  const { address, chain } = useAccount();
-  const isChainSupported = useChainIsSupported(chain?.id);
+  return (
+    <DaimoPayButtonCustom {...props}>
+      {({ show }) => (
+        <ResetContainer
+          $useTheme={theme ?? context.theme}
+          $useMode={mode ?? context.mode}
+          $customTheme={customTheme ?? context.customTheme}
+        >
+          <ThemeContainer
+            onClick={() => {
+              if (onClick) {
+                onClick(show);
+              } else {
+                show();
+              }
+            }}
+          >
+            <ThemedButton
+              theme={theme ?? context.theme}
+              mode={mode ?? context.mode}
+              customTheme={customTheme ?? context.customTheme}
+            >
+              <DaimoPayButtonInner />
+            </ThemedButton>
+          </ThemeContainer>
+        </ResetContainer>
+      )}
+    </DaimoPayButtonCustom>
+  );
+}
 
-  const { data: ensName } = useEnsName({
-    chainId: 1,
-    address: address,
-  });
+/** Like DaimoPayButton, but with custom styling. */
+function DaimoPayButtonCustom(props: DaimoPayButtonCustomProps) {
+  const isMounted = useIsMounted();
+  const context = usePayContext();
 
   // Pre-load payment info in background.
-  const { setPayId } = context.paymentInfo;
-  useEffect(() => {
-    setPayId(payId);
-  }, [payId]);
-
-  const hide = () => context.setOpen(false);
+  const { paymentState } = context;
+  useEffect(
+    () => {
+      if ("payId" in props) {
+        paymentState.setPayId(props.payId);
+      } else if ("appId" in props) {
+        paymentState.setPayParams(props);
+      } else {
+        console.error(`[BUTTON] must specify either payId or appId`);
+      }
+    },
+    "payId" in props
+      ? [props.payId]
+      : [
+          // Use JSON to avoid reloading every render
+          // eg. given paymentOptions={array literal}
+          JSON.stringify([
+            props.appId,
+            props.nonce,
+            props.toChain,
+            props.toAddress,
+            props.toToken,
+            "" + props.toAmount,
```

Comment:
this entire file reads quite confusing, recommend typing better / `useMemo`s for some of this stuff?